### DISCUSSION TITLE
How to use Uno.WinUI in multi-target library

### DISCUSSION THREAD

**Discussion by PetarSlavov** [unoplatform/uno#8752] [Q&A]
I have a project with Windows and WASM heads and class library. I need to reference some types from Uno.WinUI. This works on WASM head but throws a runtime exception on Windows:

Could not load file or assembly 'Uno.UI, Version=255.255.255.255

I'm trying to conditionally reference the Uno.WinUI from the class library, but cannot or don't know how to do it properly.

**Comment by jeromelaban** [comment]
See this documentation for creation multi-targeted libraries: https://platform.uno/docs/articles/guides/how-to-create-control-libraries.html

**Comment by ToddGlodek** [comment]
This MIGHT be related to the same issue I was having...   [BLOCKER : Converting/Repurposing UWP Shared Project in WinUI3 project (XamlParseException / NavigationView)](https://github.com/unoplatform/uno/discussions/8880)  @jeromelaban  just committed the fix, a few hours ago.  Edit the project file and pay attention to the packages that get included.  There is an ERROR related to the selected TargetFramework :

```
  <ItemGroup Condition="'$(TargetFramework)'=='net5.0-windows10.0.18362'">   **<<  Says NET5.0  should say NET6.0**   
    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.0.0" />
    <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" RuntimeFrameworkVersion="10.0.22000.24" />
    <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" TargetingPackVersion="10.0.22000.24" />
  </ItemGroup>
```
